worlds/noita/Rules.py

```diff
@@ -66,9 +71,9 @@ def has_orb_count(state: CollectionState, player: int, amount: int) -> bool:
     return state.item_count("Orb", player) >= amount
 
 
-def forbid_items_at_location(multiworld: MultiWorld, location_name: str, items: Set[str], player: int):
-    location = multiworld.get_location(location_name, player)
-    GenericRules.forbid_items_for_player(location, items, player)
+def forbid_items_at_location(world: NoitaWorld, location_name: str, items: Set[str]):

```

Comment:
I think it'd make more sense to either just directly pass the location to this method or remove this function all together personally, but this one isn't that big of a deal. I understand the idea of wanting to segment everything apart, but you have everything so segmented that it makes it more difficult to follow what's actually happening.

Reply:
The location needs to be retrieved in every case which is what this is really facilitating. If anything more logic could be moved into this function to handle multiple locations, then code at three of the callers can be improved such as.

```py
def ban_items_from_shops(multiworld: MultiWorld, player: int) -> None:
    for location_name in Locations.location_name_to_id.keys():
        if "Shop Item" in location_name:
            forbid_items_at_location(multiworld, location_name, items_hidden_from_shops, player)
```

could be

```py
def ban_items_from_shops(multiworld: MultiWorld, player: int) -> None:
    shop_locations = [name for name in Locations.location_name_to_id.keys() if "Shop Item" in name]
    forbid_items_at_locations(multiworld, shop_locations, items_hidden_from_shops, player)
```

The two for loops for the other callers would be removed.